analyzers/src/SonarAnalyzer.Common/Helpers/PropertiesHelper.cs

```diff
@@ -43,18 +43,30 @@ public static XElement[] ParseXmlSettings(SourceText sourceText)
         public static bool ReadIgnoreHeaderCommentsProperty(IEnumerable<XElement> settings, string language) =>
             ReadBooleanProperty(settings, language, "ignoreHeaderComments");
 
-        private static bool ReadBooleanProperty(IEnumerable<XElement> settings, string language, string propertySuffix, bool defaultValue = false)
-        {
-            var propertyLanguage = language == LanguageNames.CSharp ? "cs" : "vbnet";
-            var propertyName = $"sonar.{propertyLanguage}.{propertySuffix}";
-            return settings.Any()
-                && GetPropertyStringValue(propertyName) is { } propertyStringValue
+        public static string[] ReadSourceFileInclusionsProperty(IEnumerable<XElement> settings, string language) =>
+            ReadCommaSeparatedArrayProperty(settings, language, "sonar.inclusions");
+
+        public static string[] ReadSourceFileExclusionsProperty(IEnumerable<XElement> settings, string language) =>
+            ReadCommaSeparatedArrayProperty(settings, language, "sonar.exclusions");
+
+        private static bool ReadBooleanProperty(IEnumerable<XElement> settings, string language, string propertySuffix, bool defaultValue = false) =>
+            settings.Any()
+                && GetPropertyStringValue(settings, GetPropertyName(language, propertySuffix)) is { } propertyStringValue
                 && bool.TryParse(propertyStringValue, out var propertyValue)
                 ? propertyValue
                 : defaultValue;
 
-            string GetPropertyStringValue(string propName) =>
-                settings.FirstOrDefault(s => s.Element("Key")?.Value == propName)?.Element("Value").Value;
-        }
+        public static string[] ReadCommaSeparatedArrayProperty(IEnumerable<XElement> settings, string language, string propertySuffix, params string[] defaultValue) =>
```

Comment:
I am not sure you need the `params string[] defaultValue` as if the inclusions or exclusions does not exist we just want an empty array.